dds_web/utils.py

```diff
@@ -193,6 +193,19 @@ def username_in_db(username):
     return False
 
 
+def email_return_user(email):
```

Comment:
I would probably rename this function to `user_from_email` or similar. But that's probably personal preference.

Reply:
This function is not needed at all actually, this schema already does this.
https://github.com/ScilifelabDataCentre/dds_web/blob/bba1ebf390439ce6bd1d0e39e5eabc1887fb7323/dds_web/api/schemas/user_schemas.py#L28-L44 

Just call `user_schemas.UserSchema().load({"email": email})` where ever you need to do this.

Reply:
Maybe, it's used in tests though. I would feel awkward calling the schemas in the tests just to do this query...

Reply:
But why add a new function in `dds_web/utils.py` just for tests? Personally I'd rather use something that was already in the web code.

Reply:
If we want to use it in the tests and don't want to use the UserSchema then we should move this function to a `utils.py` module in the tests directory

Reply:
Yes, I created that function yesterday with regard to the tests and also pondered where I should put it. Indeed, it is currently not used in the main code, but while writing the initial code for the endpoints, I repeatedly wanted to have such a function badly. 

Arguably, this was due to bad design practices, but is there really no possible use (outside of tests) for a function that returns the user for a given email-address without performing de-serialization at the same time like the schema does? 

So as an alternative, I would propose to add it as static method to the Email db.model class then, such that one can not only return the `User.primary_email` but also the `Email.user` ?

Reply:
I'm not sure I follow. 

The UserSchema returns a User object by first getting the row in the Email table which matches the email address. The `email_return_user` also returns a User object but with a join. You don't need this though, instead you could just do
```
def email_return_user(email):
    """Helper function to return the User for a given email"""
    email = models.Email.query.filter_by(email=email).one_or_none()
    return email.user
```
The `Email.user` as you suggest is already available: https://github.com/ScilifelabDataCentre/dds_web/blob/bba1ebf390439ce6bd1d0e39e5eabc1887fb7323/dds_web/database/models.py#L566
(This is what the schema uses.)

Reply:
So I would rather just use the schema than introduce a new function, but if no one agrees, then I'm fine with having the function (it's not a deal breaker) but in that case I want the function to be moved to `dds_web/tests/utils.py` 👍🏻

Reply:
> The `UserSchema` returns a `User` object by first getting the row in the `Email` table which matches the email address. The `email_return_user` also returns a `User` object but with a join. You don't need this though, instead you could just do
> 
> ```
> def email_return_user(email):
>     """Helper function to return the User for a given email"""
>     email = models.Email.query.filter_by(email=email).one_or_none()
>     return email.user
> ```

I am pretty sure I tried exactly this with no avail yesterday in the tests, because after trying out this and about 5 different versions derived from StackOverflow, I finally found one that worked (the one using the join). 

**But anyway: I moved it now to the tests!**